fortuna/src/chain/ethereum.rs

```diff
@@ -194,9 +197,15 @@ impl EntropyReader for PythContract {
             Ok(Some(reader::Request {
                 provider:        r.provider,
                 sequence_number: r.sequence_number,
+                block_number:    r.block_number.abs().try_into()?,
```

Comment:
To support my comment. This `abs` here is so subtle (and without docs) that can get easily missed if someone edits this code.

Reply:
Would suggest `struct BlockHeight()` with a validating constructor to make this never slip through, in general I'd advocate introducing more types and avoiding `type` in general.